pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,271 @@
+
+The following steps will flash the latest release onto the balenaOS image downloaded in the previous steps and pin the device to mentioned release commit.
+
+```bash
+$ commit=$(balena app ${app_slug} | grep COMMIT | awk '{print $2}')
+
+$ balena preload ${tmpimg} \
+    --app ${app_slug} \
+    --commit ${commit} \
+    --pin-device-to-release
+```
+
+### Create Update Media
+
+Next, a USB flash drive, an SD card or any relevant storage media needs to be prepared and flashed. This storage media will be used to update the device through the offline process. Plug in your storage media and run the following the commands.
+
+> Note: The process of flashing will delete all data stored on the storage media.
+
+```bash
+# list the devices available to flash
+$ balena util available-drives
+/dev/...
+
+# Please make sure `--drive` is really the disk you want to write to.
+$ sudo balena local flash ${tmpimg} \
+    --drive /dev/{{flash-usb-disk}}
+
+$ rm ${tmpimg}
+```
+
+### Process of Reprovisioning
+
+> Warning: For devices with internal storage, this procedure erases the internal media of your device. Hence, remove any mass storage devices containing user data.
+
+With the update media ready having the latest release of the application preloaded. Follow the device's provisioning instructions present on balenaCloud dashboard for your specific device.
+
+For example: For Raspberrypi devices, insert the recently flashed SD card and power up the device. When the process is complete, (re)connect any mass storage devices containing user data back to the device. Reconnect the device to the local air-gapped network(s). Later, use SSH to connect and inspect application logs, etc.
+
+#### Strategies to remotely update with an SD card or USB device
+
+If a device isn't locally deployed, one can ship the flashed SD cards or USB sticks/drives to a remote location. There someone can run the update by simply inserting them into the devices and booting.
+
+If the target device exists on an air-gapped or Internet restricted network, inserting [ssh](https://www.balena.io/docs/reference/OS/configuration/#sshkeys) keys during the configuration step will allow fleet managers at the remote site to connect into the device directly via OpenSSH and verify the update by examining container logs, etc.
+
+If the target device is connected via a low-bandwidth connection, it should eventually establish a connection to balenaCloud. Depending on the connection's quality, it may respond to [Web Terminal][web-terminal] commands and output container logs to the dashboard.
+
+### Update Device Registration(s)
+
+This section deals with device registration.
+
+> manually reflect the new state of the device in balenaCloud for offline devices or devices on private networks without Internet access
+
+#### Patch State Endpoint
+
+This step is strictly not required for devices with low bandwidth internet connections. Those devices would be able to update the state endpoint automatically.
```

Comment:
The word "strictly" confuses me in the sentence. Would it still mean the same if the word "strictly" was deleted?

Even then, whether or not this step is required would depend on the device having "low bandwidth", which is a loosely defined condition. As a user, how can I determine, with confidence, whether or not this step applies to me? _"Just tell me whether or not I should run this"_ kind of thing. :-)

Reply:
I suppose if users are blindly copying and pasting instructions without understanding what they do, having something like this forces some sort of consideration. Could be reworded to say that if a device never ever phones home, then this step is required (unless the user doesn't care how the device appears in the dash board at all).

Reply:
> Could be reworded to say that if a device never ever phones home, then this step is required (unless the user doesn't care how the device appears in the dash board at all).

This would definitely be clearer. 👍  For example, _assuming it is correct:_
```
#### Update the device state in the cloud

This step updates the device state in the cloud (balenaCloud or openBalena)
on behalf of the device, as if the device was online. It is only required for
devices that do not have an internet connection. If the device has an internet
connection, even if low bandwidth, this step may be skipped because the device
itself will contact the cloud to update its state.
```
Assuming it is correct, this also answers my other question as to what the `balena preload` flag might be called:
```
$ balena help preload
OPTIONS
  --update-cloud-state  Update the device state in the cloud (for offline updates)
```

Reply:
That's a really good description. I am adding it, until someone says otherwise. 
cc: @ab77 @pdcastro